posthog/hogql/parser.py

```diff
@@ -172,14 +172,16 @@ def visitSelectUnionStmt(self, ctx: HogQLParser.SelectUnionStmtContext):
                 flattened_queries.append(query)
             elif isinstance(query, ast.SelectUnionQuery):
                 flattened_queries.extend(query.select_queries)
+            elif isinstance(query, ast.Placeholder):
```

Comment:
You can simplify this case by folding it into the first condition: just replace `if isinstance(query, ast.SelectQuery):` with `if isinstance(query, (ast.SelectQuery, ast.Placeholder)):`